openff/bespokefit/utilities/molecule.py

```diff
@@ -92,12 +92,6 @@ def canonical_order_atoms(molecule: Molecule):
     atom_map = {i: rank for i, rank in enumerate(atom_order)}
 
     molecule = molecule.remap(atom_map, current_to_new=True)
-
-    if "atom_map" in molecule.properties:
-        molecule.properties["atom_map"] = {
-            atom_map[i]: j for i, j in molecule.properties["atom_map"].items()
-        }
-
     return molecule
```

Comment:
This fix should work for Toolkit 0.12.0, but will change behavior on 0.11.0 because the atom map won't be updated. I think the more robust fix is to make a copy of the atom map before remapping and then use that to produce the new mapping, something like:

```suggestion
    atom_map = {i: rank for i, rank in enumerate(atom_order)}
    original_atom_map = dict(molecule.properties["atom_map"])

    molecule = molecule.remap(atom_map, current_to_new=True)

    if "atom_map" in molecule.properties:
        molecule.properties["atom_map"] = {
            atom_map[i]: j for i, j in original_atom_map["atom_map"].items()
        }

    return molecule
```